src/turrets/Turret.cpp

```diff
@@ -94,7 +95,64 @@ void Turret::pickUpTurret() {
     texture.setAlpha(0);
 }
 
-// checks if there are any enemies in the turret's coverage area, this is not yet defined
+/**
+ * Checks if there are any enemies in the turret's coverage area.
+ * Jamie, March 1
+ * Revised by Rob, March 5
+ */
 bool Turret::targetScanTurret() {
+    //Get map of all zombies
+    const auto& mapZombies = GameManager::instance()->getZombies();
+
+    unsigned int closestZombieId = 0;
+    float closestZombieDist = std::numeric_limits<float>::max();
+
+    // Detect zombies
+    bool detect = false;
+    for (const auto& item : mapZombies)
+    {
+        const auto& zombie = item.second;
+        float zombieX = zombie.getX();
+        float zombieY = zombie.getY();
+
+        float xDelta = abs((abs(zombieX - ZOMBIE_WIDTH / 2) - abs(getX() - TURRET_WIDTH / 2)));
+        float yDelta = abs((abs(zombieY - ZOMBIE_HEIGHT / 2) - abs(getY() - TURRET_HEIGHT / 2)));
+        xDelta *= xDelta;
+        yDelta *= yDelta;
```

Comment:
I'd rather you move these squaring multiplies into the below sqrt call and make the floats defined as const than do this here